sdks/python/scripts/generate_pydoc.sh

```diff
@@ -247,24 +247,28 @@ EOF
 
 # Build the documentation using sphinx
 # Reference: http://www.sphinx-doc.org/en/stable/man/sphinx-build.html
+# Note we cut out warnings from apache_beam.dataframe, this package uses pandas
+# documentation verbatim.
 python $(type -p sphinx-build) -v -a -E -q target/docs/source \
   target/docs/_build -c target/docs/source \
-  -w "target/docs/sphinx-build.warnings.log"
+  |& grep -E -v 'apache_beam\.dataframe.*WARNING:' \
+  |& tee "target/docs/sphinx-build.log"
 
 # Fail if there are errors or warnings in docs
-! grep -q "ERROR:" target/docs/sphinx-build.warnings.log || exit 1
-(! grep -v 'apache_beam.dataframe' target/docs/sphinx-build.warnings.log | grep -q "WARNING:") || exit 1
```

Comment:
why `apache_beam.dataframe` was specified specially here? Just curious

Reply:
I added a comment above to explain it. The issue is that for the DataFrame API we copy docstrings verbatim from the equivalent pandas methods, and these have various issues that raise warnings in sphinx